test/logging/tlogger.go

```diff
@@ -0,0 +1,358 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// TLogger is TLogger
+type TLogger struct {
+	l     *zap.Logger
+	level int
+	t     *testing.T
+	e     map[string][]interface{} // For Collect()
+	w     *testingWriter           // Only here so it can be disabled
+}
+
+// V() returns an InfoLogger from go-logr.
+//
+// This should be the main way your tests log.
+// Most frequent usage is used directly:
+//   t.V(2).Info("Something at regular level")
+// But if something computationally difficult is to be done, can do:
+//   if l := t.V(8); l.Enabled() {
+//     x := somethingExpensive()
+//     l.Info("logging it", "expensiveThing", x)
+//   }
+//
+// Elsewhere in this documentation refers to a hypothetical .V(errorLevel) to simplify explanations. The V() function cannot write to the error level; the Error, ErrorIfErr, Fatal, and FatalIfErr methods are the only way to write to the error level.
+func (o *TLogger) V(level int) logr.InfoLogger {
```

Comment:
Is `tl` a better abbreviation for `TLogger` than `o`?

Reply:
It will get confusing with the `t` and `l` already existing.

Reply:
If you change `l` to `logger` it will be less confusing :)